docs/proposals/federation.md

```diff
@@ -404,36 +494,107 @@ These are often left implicit by customers, but are worth calling out explicitly
 1. Software failure isolation between Kubernetes clusters should be
    retained as far as is practically possible.  The federation system
    should not materially increase the failure correlation across
-   clusters.  For this reason the federation system should ideally be
-   completely independent of the Kubernetes cluster control software,
-   and look just like any other Kubernetes API client, with no special
-   treatment.  If the federation system fails catastrophically, the
-   underlying Kubernetes clusters should remain independently usable.
+   clusters.  For this reason the federation control plane software
+   should ideally be completely independent of the Kubernetes cluster
+   control software, and look just like any other Kubernetes API
+   client, with no special treatment.  If the federation control plane
+   software fails catastrophically, the underlying Kubernetes clusters
+   should remain independently usable.
 1. Unified monitoring, alerting and auditing across federated Kubernetes clusters.
 1. Unified authentication, authorization and quota management across
    clusters (this is in direct conflict with failure isolation above,
    so there are some tough trade-offs to be made here).
 
-## Proposed High-Level Architecture
-
-TBD: All very hand-wavey still, but some initial thoughts to get the conversation going...
+## Proposed High-Level Architectures
+
+Two distinct potential architectural approaches have emerged from discussions
+thus far:
+
+1. An explicitly decoupled and hierarchical architecture, where the
+Federation Control Plane sits logically above a set of independent
+Kubernetes clusters, each of which is unaware of the other clusters,
```

Comment:
It's not necessarily the case that clusters need to be unaware of each other, nor of the Federation Control Plane, in this model. By comparison, Kubelets are aware of the cluster control plane (though not of each other).